### PR TITLE
Updates to Rook requirements

### PR DESCRIPTION
**Pull request by paigecalvert**
Story: https://app.shortcut.com/replicated/story/61207/add-storage-requirements
Related thread: https://replicated.slack.com/archives/C9845152R/p1668190576553319

Previews: 
- System Requirements topic: https://63729d555309e00082796ea3--kurlsh-staging.netlify.app/docs/install-with-kurl/system-requirements
- Rook topic: https://63729d555309e00082796ea3--kurlsh-staging.netlify.app/docs/add-ons/rook/#block-storage (separated out the filesystem storage info into a parallel section: https://63729d555309e00082796ea3--kurlsh-staging.netlify.app/docs/add-ons/rook/#filesystem-storage)

### CODE REVIEW

File: src/markdown-pages/add-ons/rook.md [hunk 1]
```diff
@@ -36,11 +30,17 @@ flags-table
...
-For production clusters, Rook should be configured to use block devices rather than the filesystem.
-Enabling block storage is required with version 1.4.3+. Therefore, the `isBlockStorageEnabled` option will always be set to true when using version 1.4.3+.
-The following spec enables block storage for the Rook add-on and automatically uses disks matching the regex `/sd[b-z]/`.
-Rook will start an OSD for each discovered disk, which could result in multiple OSDs running on a single node.
-Rook will ignore block devices that already have a filesystem on them.
+For Rook versions 1.4.3 and later, block storage is required.
+For Rook versions earlier than 1.4.3, block storage is recommended in production clusters.
+
+You can enable and disable block storage for Rook with the `isBlockStorageEnabled` field in the kURL spec.
+By default, the `isBlockStorageEnabled` field is set to `true` for Rook versions 1.4.3 and later.
```
Comment (emosbaugh) [inline src/markdown-pages/add-ons/rook.md:37]: that second section seems implicit from above. perhaps there is a better way to word it?

```suggestion
You can enable and disable block storage for Rook versions earlier than 1.4.3 with the `isBlockStorageEnabled` field in the kURL spec.
```

File: src/markdown-pages/add-ons/rook.md [hunk 2]
```diff
@@ -36,11 +30,17 @@ flags-table
...
+
+When the `isBlockStorageEnabled` field is set to `true`, Rook starts an OSD for each discovered disk.
```
Comment (emosbaugh) [inline src/markdown-pages/add-ons/rook.md:39]:
```suggestion
When the `isBlockStorageEnabled` field is set to `true`, or when using Rook versions 1.4.3 and later, Rook starts an OSD for each discovered disk.
```

File: src/markdown-pages/add-ons/rook.md [hunk 3]
```diff
@@ -50,8 +50,24 @@ spec:
     blockDeviceFilter: sd[b-z]
 ```
 
-The Rook add-on will wait for a disk before continuing.
-If you have attached a disk to your node but the installer is still waiting at the Rook add-on installation step, refer to the [troubleshooting guide](https://rook.io/docs/rook/v1.0/ceph-common-issues.html#osd-pods-are-not-created-on-my-devices) for help with diagnosing and fixing common issues.
+In the example above, the `isBlockStorageEnabled` field is set to `true`.
+Additionally, `blockDeviceFilter` instructs Rook to use only block devices that match the specified regex.
+For more information about the available options, see [Advanced Install Options](#advanced-install-options) above.
+
+The Rook add-on waits for a disk before continuing with installation. If you attached a disk to your node, but the installer is waiting at the Rook add-on installation step, see [OSD pods are not created on my devices](https://rook.io/docs/rook/v1.0/ceph-common-issues.html#osd-pods-are-not-created-on-my-devices) in the Rook documentation for troubleshooting information.
```
Comment (emosbaugh) [inline src/markdown-pages/add-ons/rook.md:57]:
```suggestion
The Rook add-on waits for a disk before continuing with installation.
If you attached a disk to your node, but the installer is waiting at the Rook add-on installation step, see [OSD pods are not created on my devices](https://rook.io/docs/rook/v1.0/ceph-common-issues.html#osd-pods-are-not-created-on-my-devices) in the Rook documentation for troubleshooting information.
```

File: src/markdown-pages/add-ons/rook.md [hunk 4]
```diff
@@ -50,8 +50,24 @@ spec:
...
+
+## Filesystem Storage
+
+By default, for Rook versions earlier than 1.4.3, the cluster uses the filesystem for Rook storage.
+However, block storage is recommended for Rook in production clusters. For more information, see [Block Storage](#block-storage) above.
```
Comment (emosbaugh) [inline src/markdown-pages/add-ons/rook.md:62]:
```suggestion
However, block storage is recommended for Rook in production clusters.
For more information, see [Block Storage](#block-storage) above.
```

File: src/markdown-pages/add-ons/rook.md [hunk 5]
```diff
@@ -50,8 +50,24 @@ spec:
...
+
+When using the filesystem for storage, each node in the cluster has a single OSD backed by a directory in `/opt/replicated/rook`.
+Nodes with a Ceph Monitor also use `/var/lib/rook`.
+
+At minimum, 10GB of disk space must be available to `/var/lib/rook` for the Ceph Monitors and other configs.
+We recommend a separate partition to prevent a disruption in Ceph's operation as a result of `/var` or the root partition running out of space.
+
+**Note**: All disks used for storage in the cluster should be of similar size. A cluster with large discrepancies in disk size may fail to replicate data to all available nodes.
```
Comment (emosbaugh) [inline src/markdown-pages/add-ons/rook.md:70]:
```suggestion
**Note**: All disks used for storage in the cluster should be of similar size.
A cluster with large discrepancies in disk size may fail to replicate data to all available nodes.
```

File: src/markdown-pages/add-ons/rook.md [hunk 6]
```diff
@@ -50,8 +50,24 @@ spec:
...
+
+## Filesystem Storage
+
+By default, for Rook versions earlier than 1.4.3, the cluster uses the filesystem for Rook storage.
+However, block storage is recommended for Rook in production clusters. For more information, see [Block Storage](#block-storage) above.
+
+When using the filesystem for storage, each node in the cluster has a single OSD backed by a directory in `/opt/replicated/rook`.
+Nodes with a Ceph Monitor also use `/var/lib/rook`.
+
+At minimum, 10GB of disk space must be available to `/var/lib/rook` for the Ceph Monitors and other configs.
```
Comment (emosbaugh) [inline src/markdown-pages/add-ons/rook.md:67]: Where did we come up with this number? I will assume its correct.

File: src/markdown-pages/add-ons/rook.md (hunk 1 above)
Comment (paigecalvert) [reply src/markdown-pages/add-ons/rook.md:37]: @emosbaugh good catch. I think my question here was that it wasn't clear to me if the `isBlockStorageEnabled` field only exists in versions earlier than 1.4.3 (if so, I can update the flags table to make this more explicit)

File: src/markdown-pages/add-ons/rook.md (hunk 6 above)
Comment (paigecalvert) [reply src/markdown-pages/add-ons/rook.md:69]: I am not sure. This was here previously, so also assumed it was correct.

File: src/markdown-pages/add-ons/rook.md (hunk 2 above)
Comment (paigecalvert) [reply src/markdown-pages/add-ons/rook.md:39]: Same question as above: It sounds like it's _not_ that the `isBlockStorageEnabled` field is set to `true` by default in 1.4.3 and later, but rather that the field no longer exists in 1.4.3 and later. Is that right?

File: src/markdown-pages/add-ons/rook.md (hunk 1 above)
Comment (emosbaugh) [reply src/markdown-pages/add-ons/rook.md:37]: yes isBlockStorageEnabled only exists for earlier versions

**Comment by paigecalvert** [comment]
Thanks @emosbaugh , committed your suggestions and this should be ready for another review

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**